core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/WebAppsDefaultHeartbeatProvider.java

```diff
@@ -0,0 +1,147 @@
+  /**
+   * Name of the provider
+   */
+  private final String name = "webapps";
+
+  /**
+   * Collection holding default properties for this default provider.
+   */
+  private final Set<String> defaultFields;
+
+  /**
+   * Map for storing environment variables
+   */
+  private Map<String, String> environmentMap;
+
+  /**
+   * Constructor that initializes fields and load environment variables
+   */
+  public WebAppsDefaultHeartbeatProvider() {
+    defaultFields = new HashSet<>();
+    environmentMap = System.getenv();
+    initializeDefaultFields(defaultFields);
+  }
+
+  @Override
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public boolean isKeyWord(String keyword) {
+    return defaultFields.contains(keyword);
+  }
+
+  @Override
+  public Callable<Boolean> setDefaultPayload(final List<String> disableFields,
+      final HeartBeatProviderInterface provider) {
+    return new Callable<Boolean>() {
+      volatile boolean hasSetValues = false;
+      volatile Set<String> enabledProperties = MiscUtils.except(defaultFields, disableFields);
+      @Override
+      public Boolean call() {
+
+        //update environment variable to account for
+        updateEnvironmentVariableMap();
+        for (String fieldName : enabledProperties) {
+          try {
+            switch (fieldName) {
+              case "website_site_name":
+                String webSiteName = getWebsiteSiteName();
+                if (webSiteName == null) {
+                  InternalLogger.INSTANCE.trace("Web site name not available, probably not a web app");
+                  break;
+                }
+                provider.addHeartBeatProperty(fieldName, webSiteName, true);
+                hasSetValues = true;
+                break;
+              case "website_home_name":
+                String webSiteHostName = getWebsiteHostName();
+                if (webSiteHostName == null) {
+                  InternalLogger.INSTANCE.trace("web site host name not available, probably not a web app");
+                  break;
+                }
+                provider.addHeartBeatProperty(fieldName, webSiteHostName, true);
+                hasSetValues = true;
+                break;
+              default:
+                InternalLogger.INSTANCE.trace("Unknown web apps property encountered");
+                break;
+            }
+          }
+          catch (Exception e) {
+            InternalLogger.INSTANCE.warn("Failed to obtain heartbeat property, stack trace"
+                + "is: %s", ExceptionUtils.getStackTrace(e));
+          }
+        }
+        return hasSetValues;
+      }
+    };
+  }
+
+  /**
+   * Populates the default Fields with the properties
+   * @param defaultFields
+   */
+  private void initializeDefaultFields(Set<String> defaultFields) {
+    if (defaultFields == null) {
+      defaultFields = new HashSet<>();
+    }
+
+    defaultFields.add("website_site_name");
+    defaultFields.add("website_home_name");
```

Comment:
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/c61fb45b5d31f5910320ea9d788de528e16d8ff0/Src/WindowsServer/WindowsServer.Shared/AppServicesHeartbeatTelemetryModule.cs#L21

Should we use the same set used by .NET?
If you are using slot, then it need to be periodically refreshed as well.

Reply:
Yes I need to change the names to match .NET. Unfortunately this were added recently in .NET, not when I implemented this.

Reply:
This is addressed.